### PR TITLE
Implements linking of outputdatastreams in the ui

### LINKED ISSUE #519: OutputDataStreams UI graph support
Related to output data streams - #224

This story is about adding desktop UI support for such jobs. Things to cover (break down if necesary):

 - [x] Drawing of the OutputDataStreamJobs on the canvas (assuming a job is opened which already has one or more "melons").
 - [x] Right clicking on a component that has output data streams should yield a new option: "Link [stream name] to ..."
 - [x] Linking a new component to a OutputDataStreamJob. This involves "moving" it from the parent job to a child job builder.

### PR DESCRIPTION
**Pull request by LosD**
This makes it possible to link a component that has an outputdatastream to other components. This is in no way a complete implementation of ways to link (i.e. it is not possible to link a component using the column list).

Fixes #519

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLinkPainter.java [hunk 1]
```diff
@@ -201,7 +229,18 @@ private boolean createLink(final Object fromVertex, final Object toVertex, final
...
+
                 try {
+                    if (sourceAnalysisJobBuilder != componentBuilder.getAnalysisJobBuilder()){
+                        if(componentBuilder.getInput().length > 0) {
+                            logger.debug("createLink(...) returning false - cannot change analysisJobBuilder for a component with configured inputColumns");
+                            return false;
+                        }
+                        componentBuilder.getAnalysisJobBuilder().removeComponent(componentBuilder);
+                        sourceAnalysisJobBuilder.addComponent(componentBuilder);
+                        componentBuilder.setAnalysisJobBuilder(sourceAnalysisJobBuilder);
```
Comment (kaspersorensen) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLinkPainter.java]: Maybe avoid this "remove + add + set" sequence have a single method (in either AnalysisJobBuilder or ComponentBuilder) that "moves" the component to a different scope?

**Comment by kaspersorensen** [comment]
Mostly looks good to me, so I am ok with merging. My only hesitation is that this is the first PR that really "exposes" the output data streams to the end users and since other features are still dangling it also exposes bugs that was previosly harmless.

Maybe we should at least first create 4.0.x branch from the master _before_ merging this. That way we can still maintain 4.0.x releases while building version 4.1 with output data streams.

**Comment by LosD** [comment]
Is there any reason to do that before problems arrive? The only change is that we'll be porting all non-melon changes back.

We can always make the branch later from the last non-melon commit, and then cherry pick the changes we want.

**Comment by LosD** [comment]
Fixed comment. Good catch, no need to make the JobGraphLinkPainter have even more non-link-painting to do :)

File: api/src/main/java/org/datacleaner/job/output/OutputDataStreamBuilderImpl.java [hunk 2]
```diff
@@ -33,7 +35,10 @@
 
     public OutputDataStreamBuilderImpl(String name) {
         _name = name;
-        _table = new MutableTable(name);
+        MutableSchema schema = new MutableSchema();
+        schema.setName(null);
+        _table = new MutableTable(name, schema);
+        schema.addTable(_table);
```
Comment (LosD) [inline api/src/main/java/org/datacleaner/job/output/OutputDataStreamBuilderImpl.java]: Damn, this actually belonged in the now-merged fix for #563

**Comment by LosD** [comment]
Merging after oral agreement.

**Pull request merged**